### DISCUSSION TITLE
Document encrypted connection to JSON-RPC API

### DISCUSSION THREAD

**Discussion by ann0see** [jamulussoftware/jamulus#3422] [Documentation]
I've played around a bit with the JSON-RPC API in the past days.
It makes sense to - for now until Jamulus supports encrypted connection of the JSON-RPC server - document remote access via e.g. Websockets. 

I came across websockify by noVNC: https://github.com/novnc/websockify
If a ssl certificate e.g via letsencrypt is available, it's quite easy to set up a proxy:
```
 sudo websockify --cert /path/to/cer --key /path/to/key --ssl-only external:<port> localhost:<jsonrpcport> 
```

Jamulus' RPC server would listen to localhost:<jsonrpcport> and then websockify adds the encrypted layer as websocket. 
Probably there are other ways to achieve this too - for my use case this seemed to be the best way.

**Comment by rdica** [comment]
@ann0see Can you detail your use case?

I use scripts over ssh. I also have a zenity "gui" that utilizes scripts over ssh.

**Reply by ann0see** [reply]
I am working on an iOS app in swift to control the server over JSON-RPC.
